### ISSUE TITLE
Option for tracking bottomless compost bucket as consuming 1/2 ultracompost each use/fill

### ISSUE THREAD

**Issue by daboross** [MosheBenZacharia/GP-Per-Hour#55]
Hi!

Is there any chance you'd be open to changing how the compost bucket works, or adding an optional mode changes it?

I feel like the current way of doing it shows farm runs as slightly less profitable, and compost runs significantly more profitable than they are.

If I buy 100 ultracompost on the GE, then fill my compost bucket with it, GP/hour tracks this action as +100 ultracompost:
![image](https://github.com/user-attachments/assets/a1d9e099-64d9-44cb-a4c1-a88999e82411)

Later, when I do a farm run, plant 14 seeds, and compost them all, it tracks this as -14 buckets of ultracompost:

![image](https://github.com/user-attachments/assets/6a11bdc6-1531-4a33-b238-1642bd1959bf)

It looks like I spent 6104 gold on compost for that run, but since replenishing that compost only cost me 7 buckets from the GE, 3052 gold would more accurate.

The doubling is accounted for when I refill my bucket in bulk, but I'd ideally like to do that infrequently - making it difficult to accurately track each farming trip standalone.

---

For compost runs, consider this trip, harvesting and refilling 1 regular compost bin:

![image](https://github.com/user-attachments/assets/93a95808-56d1-4bde-8478-21813e9736cc)

It looks like I've just made 290k GP/hour, but when compared to buying compost off of the GE, this only made me 145k GP/hour, or slightly less.

**Comment by MosheBenZacharia** [comment]
That makes sense. So the solution would be to count the charges as half a bucket, is that right?

So if your bucket has 1k ultra compost charges that would count as 500 ultra compost?

This would make it so that adding compost to the bucket does not register any profit, and removing one compost would count as a 0.5 compost loss. Does that seem correct?

**Comment by daboross** [comment]
Yep, that would be correct! It'd work great for me, and I assume it would be more generally useful as well.

**Comment by MosheBenZacharia** [comment]
ok cool, that sounds good. I have a PR thats been open for like 3 weeks now so I'm prob gonna wait for that to merge before adding this change since it would move me to the back of the review queue.

**Comment by MosheBenZacharia** [comment]
I added this functionality in 3b0628a45f4ca2005672e970000327a11ab8b472 . will hopefully get this in a PR soon

**Comment by MosheBenZacharia** [comment]
should be fixed when this gets merged: https://github.com/runelite/plugin-hub/pull/7310

**Comment by MosheBenZacharia** [comment]
should be fixed in 1.10
